gems/pending/spec/postgres_ha_admin/failover_databases_spec.rb

```diff
@@ -1,132 +1,93 @@
+    it "return list of active databases saved in 'config/failover_databases.yml'" do
+      File.write(@yml_file, all_db_list.to_yaml)
+      expect(@failover_databases.active_databases).to contain_exactly(
+        {:type => 'master', :active => true, :host => '1.1.1.1', :user => 'root', :dbname => 'vmdb_test'},
+        {:type => 'standby', :active => true, :host => '2.2.2.2', :user => 'root', :dbname => 'vmdb_test'})
+    end
+  end
 
-      connection.exec(<<-SQL)
-        CREATE TABLE #{described_class::TABLE_NAME} (
+  describe "#update_failover_yml" do
+    before do
+      # @connection = PG::Connection.open(:dbname => 'vmdb_test')
+      begin
+        @connection = PG::Connection.open(:dbname => 'travis', :user => 'travis')
+      rescue PG::ConnectionBad => _err
+        skip "travis database does not exist"
+      end
+
+      @connection.exec("START TRANSACTION")
+      @connection.exec(<<-SQL)
+        CREATE TABLE #{described_class::TABLE_NAME}  (
           type text NOT NULL,
           conninfo text NOT NULL,
           active boolean DEFAULT true NOT NULL
         )
       SQL
 
-      connection.exec(<<-SQL)
+      @connection.exec(<<-SQL)
         INSERT INTO
           #{described_class::TABLE_NAME}(type, conninfo, active)
         VALUES
           ('master', 'host=1.1.1.1 user=root dbname=vmdb_test', 'true'),
           ('standby', 'host=2.2.2.2 user=root dbname=vmdb_test', 'true'),
           ('standby', 'host=3.3.3.3 user=root dbname=vmdb_test', 'false')
       SQL
-
-      example.run
-    ensure
-      connection.exec("ROLLBACK")
-      connection.finish
-      remove_file
     end
-  end
-
-  describe "#all_databases" do
-    it "returns list of all available databases saved in yaml file" do
-      list = subject.all_databases(connection)
 
-      expect(list.size).to be 3
-      expect(list).to include({:type => "master", :host => "1.1.1.1", :dbname => "vmdb_test",
-                               :user => "root", :active => true},
-                              {:type => "standby", :host => "2.2.2.2", :dbname => "vmdb_test",
-                               :user => "root", :active => true},
-                              {:type => "standby", :host => "3.3.3.3", :dbname => "vmdb_test",
-                               :user => "root", :active => false})
+    after do
+      @connection.exec("ROLLBACK")
+      @connection.finish
     end
 
-    it "create yaml file with list of available databases if file did not exist" do
-      expect(File.exist?(subject.yml_file)).to be false
+    it "updates 'failover_databases.yml'" do
+      @failover_databases.update_failover_yml(@connection)
 
-      subject.all_databases(connection)
-      expect(File.exist?(subject.yml_file)).to be true
-    end
-
-    it "if yaml file exists, it loads list of databases from existing file" do
-      list = subject.all_databases(connection)
-      expect(list.size).to be 3
+      expect(@failover_databases.active_databases).to contain_exactly(
```

Comment:
This should probably be testing the contents of the file rather than the return value of another method.

A yaml load and hash comparison would probably be better, especially because we are expecting the file to also contain the not active databases right?